src/app/shared/code-editor-list/code-editor-list.service.ts

```diff
@@ -68,23 +60,13 @@ export class CodeEditorListService {
    * @returns {boolean}
    * @memberof CodeEditorListService
    */
-  static isPrimaryDescriptor(path: string | null): boolean {
+  isPrimaryDescriptor(path: string | null): boolean {
     if (!path) {
       return false;
     }
-    const primaryDescriptors = [
-      // SMK
-      '/Snakefile',
-      // CWL
-      '/Dockstore.cwl',
-      // WDL
-      '/Dockstore.wdl',
-      // NFL
-      CodeEditorListService.NEXTFLOW_CONFIG_PATH,
-      CodeEditorListService.NEXTFLOW_PATH,
-      // Galaxy
-      '/Dockstore.yml',
-    ];
+    const primaryDescriptors = this.descriptorLanguageService.getDescriptorLanguagesDefaultDescriptorPaths();
+    // CodeEditorListService.NEXTFLOW_PATH isn't currently in Nextflow.ts/extendedDescriptorLanguage
```

Comment:
It would be nice if you didn't have to special case this, but I don't entirely understand the problem.

Reply:
added a comment.

Reply:
`multiple primary descriptors` if I understand correctly

Reply:
Yes, and this code could be simplified by accommodating multiple descriptors in ExtendedDescriptorLanguageBean; but should probably be a separate PR.